src/main/java/net/imagini/zkcli/ZkCli.java

```diff
@@ -96,99 +72,25 @@ private void doCliActions() {
         }
     }
 
-    private Optional<CuratorFramework> getCluster(CliParameters cliParameters) {
-        try {
-            return Optional.ofNullable(cliParameters.cluster == null || cliParameters.cluster.isEmpty()
-                    ? ZkClusterManager.buildClient(cliParameters.zkConnect)
-                    : ZkClusterManager.getClient(cliParameters.cluster));
-        } catch (RuntimeException ex) {
-            return Optional.empty();
-        }
-    }
-
-    private void printPathData(CuratorFramework client, String path) {
-        try {
-            System.out.println(new String(client.getData().forPath(path), CHARSET));
-        } catch (Exception e) {
-            throw new RuntimeException(e);
-        }
-    }
-
-    private void printPathMetaData(CuratorFramework client, String path, String metadataItemGetterName) {
-        try {
-            System.out.println(formatMetaData(client.checkExists().forPath(path), metadataItemGetterName));
-        } catch (Exception e) {
-            throw new RuntimeException(e);
+    private void runCliActionsForPath(CuratorFramework client, String path) {
+        if (params.isListChildren()) {
```

Comment:
I did think about something like that. I'd also considered something more like a typical pipeline where you construct a list of partially applied consumers so the only argument is the path to operate on then send each path to each consumer in order sort of like a pipe-lined pub sub system. I'd be happy with either option though similarly I'm feeling this is already a pretty big commit, I changed what I needed to in order to make it testable but general refactorings like that should probably be a separate commit.